code/modules/projectiles/guns/projectile/sniper.dm

```diff
@@ -18,14 +18,37 @@
 	reload_sound 	= 'sound/weapons/guns/interact/rifle_load.ogg'
 	matter = list(MATERIAL_PLASTEEL = 40, MATERIAL_PLASTIC = 20)
 	var/bolt_open = 0
+	zoom_factor = 2.0
 
 /obj/item/weapon/gun/projectile/heavysniper/update_icon()
 	if(bolt_open)
 		icon_state = "heavysniper-open"
 	else
 		icon_state = "heavysniper"
 
-/obj/item/weapon/gun/projectile/heavysniper/attack_self(mob/user as mob)
+/obj/item/weapon/gun/projectile/heavysniper/attack_self(mob/user) //Someone overrode attackself for this class, soooo.
+	if(zoom)
+		toggle_scope(zoom_factor)
+		return
+	var/list/options = list("firemode", "scope", "safety")
+	for(var/option in options)
+		options[option] = image(icon = 'icons/obj/gun_actions.dmi', icon_state = "[option]")
+	var/selected
+	selected = show_radial_menu(user, src, options, radius = 42)
+	if(!selected)
+		return
+	switch(selected)
+		if("firemode")
+			var/datum/firemode/new_mode = switch_firemodes(user)
+			if(new_mode)
+				playsound(src.loc, 'sound/weapons/guns/interact/selector.ogg', 100, 1)
+				user << SPAN_NOTICE("\The [src] is now set to [new_mode.name].")
```

Comment:
Same here.

Reply:
Didnt know you had that 🥙

Reply:
It was a thing for a good while, but it wasn't enforced. I just added that to the contributing guide.